src/NUnitTestExtractor/NUnitTestExtractorApp.cs

```diff
@@ -16,85 +16,66 @@ public static class NUnitTestExtractorApp
     {
         private static Options _options = new Options();
 
-        public enum Level { Namespace, Class, Function, TestCase, Null }
+        public enum Level { Namespace, Class, Function, TestCase };
 
-        static void Main(string[] args)
+        public static void Main(string[] args)
         {
-            //Setting a default value
-            Level level = Level.Null;
-
             using (var parser = new CommandLine.Parser(with => with.HelpWriter = Console.Error))
             {
                 if (parser.ParseArgumentsStrict(args, _options, () => Environment.Exit(-1)))
                 {
                     if (_options.Help)
                     {
                         Console.WriteLine(CommandLine.Text.HelpText.AutoBuild(_options));
+
                         return;
                     }
-                   
-                    if(!string.IsNullOrEmpty(_options.Output))
-                    {
-                        string directory = null;
 
-                        int index = _options.Output.LastIndexOf("/", StringComparison.OrdinalIgnoreCase);
-
-                        if (index > 0)
-                        {
-                            directory = _options.Output.Substring(0, index);
-                        }  
+                    // _options.Output is an abusolutely path of output file for the test name list
+                    if (!string.IsNullOrEmpty(_options.Output))
```

Comment:
[Could] You could move this if-block into a private CreateOutputFile() function.

Reply:
Yep, all detail implementation in main will be abstracted into private function. You will see that in the final version.